sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectFilterImpl.java

```diff
@@ -0,0 +1,53 @@
+package life.qbic.portal.sampletracking.components.projectoverview;
+
+import static org.apache.commons.lang.StringUtils.containsIgnoreCase;
+
+import java.util.function.Predicate;
+
+/**
+ * <b>A filter for project summaries</b>
+ *
+ * <p>This filter can be used to filter project summary objects. It can be configured and later retrieved as
+ * a predicate.</p>
+ *
+ * <pre>
+ * {@code
+ * var condition = new ProjectFilterImpl().containingText("my text").asPredicate();
+ * projects.filter(condition); // only project summaries with "my text" in their title or code
+ * }
+ * </pre>
+ *
+ * @since 1.0.0
+ */
+public class ProjectFilterImpl implements ProjectFilter {
+
+  private String substring = "";
+  private boolean emptyProjectAllowed = false;
+
+  @Override
+  public ProjectFilter containingText(String substring) {
+    this.substring = substring;
+    return this;
+  }
+
+  @Override
+  public ProjectFilter allowEmptyProjects(boolean emptyProjectAllowed) {
+    this.emptyProjectAllowed = emptyProjectAllowed;
+    return this;
+  }
+
+  @Override
+  public Predicate<? extends ProjectSummary> asPredicate() {
+
+    return containsText().and(it -> emptyProjectAllowed || it.getTotalSampleCount() > 0);
+  }
```

Comment:
What is an empty project?
If I recall correctly total sample count > 0 is currently set directly as a filter on the grid. I don't think we need to add it here. I think removing those, in general, is a precondition for the content in the gird and the other is custom filtering for what the user types in. I would prefer to keep those separate

Reply:
The previous solution always overwrote my filters. This way it is all handled by the filter.